crates/bevy_ecs/src/query/filter.rs

```diff
@@ -593,17 +593,12 @@ macro_rules! impl_tick_filter {
 }
 
 impl_tick_filter!(
-    /// Filter that retrieves components of type `T` that have been added since the last execution
-    /// of this system.
+    /// A filter on a component that only retains results added after the system last ran.
     ///
-    /// This filter is useful to do one-time post-processing on components.
+    /// A common use for this filter is one-time initialization.
     ///
-    /// Because the ordering of systems can change and this filter is only effective on changes
-    /// before the query executes you need to use explicit dependency ordering or ordered stages to
-    /// avoid frame delays.
-    ///
-    /// If instead behavior is meant to change on whether the component changed or not
-    /// [`ChangeTrackers`](crate::query::ChangeTrackers) may be used.
+    /// To retain all results without filtering but still check whether they were added after the
+    /// system last ran, use [`ChangeTrackers<T>`](crate::query::ChangeTrackers).
```

Comment:
Or use `Added<T>` in your query instead of in its filters and you will get a bool indicating whether that component was added or not since the last execution of the system.

Reply:
Oh, thanks for pointing that out. Can you share a token example to add to the docs?

Reply:
Hmm. I having trouble coming up with self-contained and not completely artificial applications.

Reply:
https://github.com/bevyengine/bevy/pull/4180 The performance gains there are due to avoiding the additional lookup in the filtered query. If you just get the changed state directly in the query the same information is right there.